pages/api/openPR.js

```diff
@@ -15,40 +15,178 @@ const GITHUB_BRANCH = process.env.GITHUB_BRANCH
+  data = JSON.stringify(values, null, 2).concat("\n");
+
+  // Parse project names using DPG naming convention for filenames
+  name = parseProjectName(values);
+
+  // Add nominee submission to nominee directory
+  nomineePath = "nominees/" + `${name}`;
+
+  // Add DPG submission to screening directory
+  dpgPath = "screening/" + `${name}`;
+
+  stage === "nominee"
+    ? Object.entries(nomineeJSON).forEach(
+        () =>
+          (files = {
+            [nomineePath]: {
+              content: nomineeJSON,
+              encoding: "utf-8",
+            },
+          })
+      )
+    : Object.entries(nomineeJSON).forEach(
+        () =>
+          (files = {
+            [nomineePath]: {
+              content: nomineeJSON,
+              encoding: "utf-8",
+            },
+            [dpgPath]: {
+              content: data,
+              encoding: "utf-8",
+            },
+          })
+      );
+
+  return files;
+}
+
+// Nominee processing before opening pull request
+function nomineeSubmission(values, sortedSubmission) {
+  // Sort entries by iterating through object and unpacking entries
+  Object.entries(values).forEach(
+    () =>
+      (sortedSubmission = {
+        name: values.name ? values.name : "",
+        aliases: values.aliases ? [values.aliases] : [""],
+        description: values.description ? values.description : "",
+        website: values.website ? values.website : "",
+        license: values.license ? values.license : [],
+        SDGs: values.SDGs ? values.SDGs : [],
+        sectors: values.sectors ? values.sectors : [],
+        type: values.type ? values.type : [],
+        repositoryURL: values.repositoryURL ? values.repositoryURL : "",
+        organizations: values.organizations ? [values.organizations] : [],
+        stage: values.stage ? values.stage : "",
+      })
+  );
+
+  // Order nominee fields in the correct order e.g organizations
+  sortedSubmission = orderFields(sortedSubmission);
+  return sortedSubmission;
+}
+
 export default async (req, res) => {
   if (req.method === "POST") {
-    const values = req.body.values;
-    console.log(values);
+    let values = req.body.values;
+
+    let nomineeJSON, sdgNumber, evidenceText, sortedSubmission;
 
     // Exclude contact information from pull request
     delete values["contact"];
 
-    let sdgNumber, evidenceText;
-    for (let i = 0; i < values.SDGs.length; i++) {
-      sdgNumber = parseInt(values.SDGs[i]);
-      evidenceText = "evidenceText".concat(sdgNumber);
-
-      values.SDGs[i] = {
-        SDGNumber: sdgNumber,
-        evidenceText: values[evidenceText],
-      };
+    // Parse SDG information
+    values = getSDGRelevanceInfo(values, sdgNumber, evidenceText);
 
-      delete values[evidenceText];
-    }
+    // Verify submission stage(nominee/ DPG) and channel to nomination or DPG review processing
+    sortedSubmission =
+      values.stage === "nominee"
```

Comment:
This ternary operator is now irrelevant if both conditions lead to the same result. This should be thus rewritten as:
`sortedSubmission = nomineeSubmission(values, sortedSubmission)`. Shouldn't it?

Reply:
Fixed in commit 9509763.